awkward/array/jagged.py

```diff
@@ -134,7 +134,10 @@ def fromiter(cls, iterable):
     @classmethod
     def fromoffsets(cls, offsets, content):
         offsets = cls._util_toarray(offsets, cls.INDEXTYPE, cls.numpy.ndarray)
-        if offsets.base is not None:
+        if hasattr(offsets.base, 'base') and str(type(offsets.base.base)) == "<class 'pyarrow.lib.Buffer'>":
```

Comment:
```python
type(offsets.base.base).__module__ == "pyarrow.lib" and type(offsets.base.base).__name__ == "Buffer"
```

would be safer. The `__name__` exists in Pythons 2 and 3 (unlike the new `__qualname__`), and the above is essentially the same as `type(offsets.base.base) == pyarrow.lib.Buffer` without having to import pyarrow.

Reply:
Cool thanks